app/services/import/legacy/transform-company-addresses.service.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+/**
+ * Transforms NALD  data into a valid object that matches the WRLS structure for a company address
+ * @module ImportLegacyTransformCompanyAddressesService
+ */
+
+const CompanyAddressPresenter = require('../../../presenters/import/legacy/company-address.presenter.js')
+const FetchCompanyAddressesService = require('./fetch-company-address.service.js')
+
+/**
+ * Transforms NALD  data into a valid object that matches the WRLS structure for a company address
+ *
+ * @param {string} regionCode - The NALD region code
+ * @param {string} licenceId - The NALD licence ID
+ * @param {object[]} transformedCompanies
+ *
+ */
+async function go (regionCode, licenceId, transformedCompanies) {
+  const naldAddresses = await FetchCompanyAddressesService.go(regionCode, licenceId)
+
+  naldAddresses.forEach((naldAddress) => {
+    const matchingCompany = _matchingCompany(transformedCompanies, naldAddress)
+
+    const address = CompanyAddressPresenter.go(naldAddress)
+
+    matchingCompany.companyAddresses = [...(matchingCompany?.companyAddresses || []), address]
```

Comment:
I'm not sure we should be playing it safe here. If we somehow got an address back from the FetchCompanyAddressesService, that didn't match to a company we'd found in an earlier step, something has gone wrong (I believe).

I'd rather it blow up and we go looking, than the issue be quietly swallowed.

Reply:
change addresses